plugins/plugin-chart-echarts/src/Graph/types.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+export type EchartsGraphFormData = {
+  name: string;
+  metric?: string;
+  zoom: number;
+  layout: string;
+  circularConfig: object;
+  forceConfig: object;
+  roam: boolean;
+  draggable: boolean;
+  edgeSymbol: object;
+  edgeSymbolSize: object;
+  itemStyle: object;
+  labelConfig: object;
+  emphasis: object;
+  selectedMode: string;
+  autoCurveness: number;
+  left: string;
+  top: string;
+  right: string;
+  bottom: string;
+  animation: boolean;
+  animationDuration: number;
+  animationEasing: string;
+  showSymbolThreshold: number;
+  tooltipConfiguration: object;
+  lineStyleConfiguration: object;
+};
+
+export const DEFAULT_FORM_DATA: EchartsGraphFormData = {
+  name: 'directed force graph',
+  zoom: 1,
+  layout: 'force',
+  circularConfig: { rotateLabel: true },
+  forceConfig: {
+    initLayout: 'circular',
+    repulsion: 50,
+    gravity: 0.05,
+    edgeLength: 200,
+    layoutAnimation: true,
+    friction: 0.6,
+  },
```

Comment:
`FormData` corresponds what we will save in each chart's metadata. The typing can also be used to enforce what is expected from the backend API, which make it easier and safer to make future changes to `transformProps`.

IMO if something is not yet available to be configured by the controls or will not be sent to the backend, then they should not appear in `FormData`.

If this is about adding a sensible default, we can maybe make them their own constants. I believe you can also borrow typing from EChartsOption to make them more robust.

Reply:
moving them to seperate 'constants' file would be better idea?

Reply:
moved them out of form_data into constants, is it ok this way? @ktmud

Reply:
Yeah, they could be their own constants either in `types.ts` or `transformProps.ts`. Or if you decide to add a `constants.ts` file, that would be fine, too.

Reply:
yeah,moving to seperate constants file sound better.

Reply:
These would probably be nice to expose as visual customization options. I tried changing repulsion to 1000, and got nice separation on the otherwise cluttered example Energy Force Layout example chart:
With current repulsion values:
![image](https://user-images.githubusercontent.com/33317356/106896543-60fd4400-66fa-11eb-8a8e-864b5db89488.png)

With repulsion set to 1000:
![image](https://user-images.githubusercontent.com/33317356/106896434-38754a00-66fa-11eb-88db-6850e2d0f829.png)